addons/payment_asiapay/models/payment_transaction.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from werkzeug import urls
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+from .const import CURRENCY_MAPPING
+from odoo.addons.payment_asiapay import utils as payment_asiapay_utils
+from odoo.addons.payment_asiapay.controllers.main import AsiaPayController
+
+_logger = logging.getLogger(__name__)
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    def _get_specific_rendering_values(self, processing_values):
+        """Override of payment to return asiapay rendering values.
+
+        Note: self.ensure_one() from `_get_processing_values`
+
+        :param dict processing_values: The generic and specific processing values of the transaction
+        :return: The dict of acquirer-specific processing values
+        :rtype: dict
+        """
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != "asiapay":
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        order_ref = "{} {}".format(self.reference, datetime.now().strftime("%m/%d/%Y %H:%M:%S"))
```

Comment:
What is this order ref? Shouldn't it be the transaction reference?

Reply:
If only have one shop then using the transaction reference is fine. However the order_ref in Asiapay is an unique field, after I reinstall the App, the transaction reference is reset therefore the payment rejected from Asiapay due to the uniqueness of the order_ref field.

Reply:
Ah ok, so it *is* the transaction reference! I could have figured that out from the computation, but, you know, the brain and its mysteries...

We have a util method in `payment/utils.py` to do just that: "singularize" the transaction reference. Only, it should be done in an override of the `_compute_reference` method so that the reference is the same in Odoo as in AsiaPay's system. Otherwise, you will run into a lot of trouble when trying the match both. See for example [how it's done in `payment_ogone`](https://github.com/odoo/odoo/blob/master/addons/payment_ogone/models/payment_transaction.py#L22-L49). You can pretty much copy-paste the method and its docstring :)